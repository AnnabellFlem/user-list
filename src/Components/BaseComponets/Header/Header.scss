@import 'src/Styles/sizes';
@import 'src/Styles/colors';

.header {
  display: flex;
  justify-content: center;

  &__title {
    font-size: $font-l;
    line-height: $lh-s;
    color: $black-100;
  }
}
