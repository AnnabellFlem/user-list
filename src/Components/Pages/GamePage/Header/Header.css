@import '../../../../Styles/colors.css';
@import '../../../../Styles/sizes.css';

.Header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  grid-area: header;
}

.Header__nav {
  margin: 0;
  width: 100%;
}

.Header__list {
  margin: 0;
  width: 100%;
}

.Header__item + .Header__item {
  margin-top: 8px;
}

@media (max-width: 920px) {
  .Header {
    background-color: var(--black-5);
    justify-content: flex-end;
    position: absolute;
    width: 100%;
    height: 100vh;
  }

  .Header--close + main {
    overflow: hidden;
  }

  .Header__menu {
    border: none;
    background: none;
    padding: 0;
    display: flex;
    position: absolute;
    top: 16px;
    right: 16px;
    z-index: 4;
  }

  .Header__nav {
    display: flex;
    align-items: center;
    height: 100%;
    z-index: 3;
    background-color: var(--black-5);
  }
}
