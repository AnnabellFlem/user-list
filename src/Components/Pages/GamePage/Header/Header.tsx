import React, { useEffect, useState } from 'react'
import './Header.css'
import { ReactComponent as CloseIcon } from '../../../../Images/svg/close.svg'
import { ReactComponent as BurgerIcon } from '../../../../Images/svg/menu.svg'
import MoneyCell from '../../../BaseComponents/MoneyCell'
import classNames from 'classnames'
import useMedia from 'use-media'
import { MEDIA_QUERY_TABLET } from '../../../../Constants/MediaQueries'
import { moneyArray } from '../../../../Services/mockData'
import { MoneyListType } from '../../../../Types'

type Props = {
  handleBtnClick: () => void
  currentNumberQuestion: number
}

const Header: React.FC<Props> = ({ handleBtnClick, currentNumberQuestion }) => {
  const isWide = useMedia(MEDIA_QUERY_TABLET)
  const [closeIcon, setCloseIcon] = useState(false)
  const [activeRound, setActiveRound] = useState(currentNumberQuestion)

  const openText = 'Open list of money levels'
  const closeText = 'Close list of money levels'

  const iconClick = () => {
    setCloseIcon(!closeIcon)
    handleBtnClick()
  }

  const defaultHeaderClassName = 'Header'
  const headerClassNames = classNames(defaultHeaderClassName, {
    [`${defaultHeaderClassName}--close`]: closeIcon,
  })

  useEffect(() => {
    setActiveRound(currentNumberQuestion)
  }, [currentNumberQuestion])

  const DesktopMenu = (array: MoneyListType) => {
    return (
      <nav className={`${defaultHeaderClassName}__nav`}>
        <ul className={`${defaultHeaderClassName}__list`}>
          {array
            .map((item, index) => {
              return (
                <li key={item} className={`${defaultHeaderClassName}__item`}>
                  <MoneyCell
                    active={activeRound === index}
                    disabled={activeRound > index}
                  >
                    {item}
                  </MoneyCell>
                </li>
              )
            })
            .reverse()}
        </ul>
      </nav>
    )
  }

  const MobileMenu = (array: MoneyListType) => {
    return (
      <>
        <button
          className={`${defaultHeaderClassName}__menu`}
          aria-label={closeIcon ? closeText : openText}
          onClick={() => iconClick()}
        >
          {closeIcon ? <CloseIcon tabIndex={0} /> : <BurgerIcon tabIndex={0} />}
        </button>
        {closeIcon && (
          <nav className={`${defaultHeaderClassName}__nav`}>
            <ul className={`${defaultHeaderClassName}__list`}>
              {array
                .map((item, index) => {
                  return (
                    <li
                      key={item}
                      className={`${defaultHeaderClassName}__item`}
                    >
                      <MoneyCell
                        active={activeRound === index}
                        disabled={activeRound > index}
                        size={'s'}
                      >
                        {item}
                      </MoneyCell>
                    </li>
                  )
                })
                .reverse()}
            </ul>
          </nav>
        )}
      </>
    )
  }

  return (
    <header className={`${headerClassNames}`}>
      {isWide ? DesktopMenu(moneyArray) : MobileMenu(moneyArray)}
    </header>
  )
}

export default Header
