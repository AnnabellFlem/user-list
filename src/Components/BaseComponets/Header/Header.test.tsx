import React from 'react'
import { shallow } from 'enzyme'
import Header from './Header'

describe('Header test', () => {
  let wrapper: any

  beforeAll(() => {
    wrapper = shallow(<Header />)
  })

  it('should to match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Component should render header', () => {
    expect(wrapper.find('header')).toHaveLength(1)
  })

  it('Component should render h2', () => {
    expect(wrapper.find('h2')).toHaveLength(1)
  })
})
