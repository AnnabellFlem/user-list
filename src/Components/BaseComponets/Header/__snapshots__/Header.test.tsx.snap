// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test should to match snapshot 1`] = `ShallowWrapper {}`;
